packages/astro/src/core/create-vite.ts

```diff
@@ -130,6 +130,10 @@ export async function createVite(
 		},
 		ssr: {
 			noExternal: [...getSsrNoExternalDeps(settings.config.root), ...thirdPartyAstroPackages],
+			// shiki is imported by Code.astro, which is no-externalized (processed by Vite).
+			// However, shiki's deps are in CJS and trips up Vite's dev SSR transform, externalize
+			// shiki to load it with node instead.
+			external: mode === 'dev' ? ['shiki'] : [],
```

Comment:
why only in dev?

Reply:
Dev mode uses Vite's SSR transform (for HMR etc), and it doesn't handle some CJS code like shiki's. In prod, there's no SSR transform since everything can be bundled as is by Rollup (and no HMR).

We could do it in prod too, but it shouldn't change anything. So I thought it would be better left as is to be handled by Vite's default, and also the user can change it if they want to.